### ISSUE TITLE
extracting error “xz: (stdin): File format not recognized”

### ISSUE THREAD

**Issue by blockspacer** [oclint/oclint#569]
### Issue Summary

issue during local build

### Environment
- Operation system name and version:

```bash
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic
```
- OCLint version:

```bash
git clone https://github.com/oclint/oclint.git
```

- How OCLint is installed: local build? prebuilt binary downloaded from github? homebrew install? others?

local build

### Reproduction Steps

```bash
git clone https://github.com/oclint/oclint.git
cd oclint
cd oclint-scripts
./make
# ERROR: extracting error “xz: (stdin): File format not recognized”
# tar: Child returned status 1
# tar: Error is not recoverable: exiting now
```

### Expected Behavior

`./make` command must to fail

### Actual Behavior

`./make` command failed with error

```
# ERROR: extracting error “xz: (stdin): File format not recognized”
# tar: Child returned status 1
# tar: Error is not recoverable: exiting now
```

Note that `sudo apt install xz-utils -y` does not help

**Comment by blockspacer** [comment]
possible solution (The -J is the flag that specifically deals with .xz files):

```bash
cd oclint/oclint-scripts
# edit clang file
nano clang

# replace 
# process.call('tar xf prebuilt.tar.xz -C ' + install_path + ' --strip-components=1')
# with
process.call('tar xf prebuilt.tar.xz -C ' + install_path + ' --strip-components=1')
```

```bash
xz --version
xz (XZ Utils) 5.2.2
liblzma 5.2.2

tar --version
tar (GNU tar) 1.29
Copyright (C) 2015 Free Software Foundation, Inc.
```

BTW `-J` flag did not help, can we switch to other file format, please?

**Comment by ryuichis** [comment]
I believe the xz file is from llvm, which is our dependency. Not sure why it failed downloading llvm for you.

**Comment by blockspacer** [comment]
I failed to build oclint due to that problem (tried multiple re-downloads and different tools - i think that xz file corrupted or chosen tools bugged). 

Now i`m using pre-built oclint like so:

```
cd ~
wget --no-check-certificate https://github.com/oclint/oclint/releases/download/v0.13.1/oclint-0.13.1-x86_64-linux-4.4.0-112-generic.tar.gz
# mirror 1: http://github.strcpy.cn/oclint/oclint/releases/download/v0.13.1/oclint-0.13.1-x86_64-linux-4.4.0-112-generic.tar.gz
# mirror 2: http://archives.oclint.org/releases/0.8/oclint-0.8.1-x86_64-linux-3.13.0-35-generic.tar.gz
tar -xzvf oclint-0.13.1-x86_64-linux-4.4.0-112-generic.tar.gz
rm -rf oclint-0.13.1-x86_64-linux-4.4.0-112-generic.tar.gz
export OCLINT_HOME=~/oclint-0.13.1
export PATH=$OCLINT_HOME/bin:$PATH
oclint -version
```

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.

**Comment by blockspacer** [comment]
Any suggestions, anyone?

**Comment by ryuichis** [comment]
@blockspacer can you download the artifacts and unarchive them from https://github.com/llvm/llvm-project/releases/tag/llvmorg-13.0.0  ?

Because essentially here are what happened for Ubuntu 20.04:
```
curl -L -o prebuilt.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/clang+llvm-13.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz
mkdir llvm-install
tar xf prebuilt.tar.xz -C  llvm-install --strip-components=1
```

I am not sure why this failed for you, as you see, nothing here really involve OCLint, it basically download the LLVM pre-built binary directly from Github's LLVM repository and then use system tar to unarchive it.

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.

**Comment by ryuichi-assistant** [comment]
This issue will be auto-closed because there hasn't been any activity for a few months. Feel free to open a new one if you still experience this problem.
